whois-nrtm4-client/src/main/java/net/ripe/db/nrtm4/client/importer/SnapshotImporter.java

```diff
@@ -0,0 +1,180 @@
+        try {
+            final MessageDigest digest = MessageDigest.getInstance("SHA-256");
+            final byte[] encodedSha256hex = digest.digest(bytes);
+            return encodeHexString(encodedSha256hex);
+        } catch (final NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Map<RpslObject, RpslObjectUpdateInfo> persistBatches(final String[] remainingRecords,
+                                                                 final AtomicInteger processedCount) {
+        return Arrays.stream(remainingRecords)
+                .parallel()
+                .map(record -> {
+                    try {
+                        return nrtm4ClientRepository.processObject(record);
+                    } catch (JsonProcessingException e) {
+                        LOGGER.error("Unable to parse record {}", record, e);
+                        throw new IllegalStateException(e);
+                    }
+                })
+                .peek(entry -> processedCount.incrementAndGet())
+                .collect(Collectors.toMap(
+                    Map.Entry::getKey,
+                    Map.Entry::getValue
+                ));
+    }
+
+    public static RpslObject getPlaceholderPersonObject() {
+        return RpslObject.parse("" +
+                "person:         Placeholder Person Object\n" +
+                "address:        RIPE Network Coordination Centre\n" +
+                "address:        P.O. Box 10096\n" +
+                "address:        1001 EB Amsterdam\n" +
+                "address:        The Netherlands\n" +
+                "phone:          +31 20 535 4444\n" +
+                "nic-hdl:        DUMY-RIPE\n" +
+                "mnt-by:         RIPE-DBM-MNT\n" +
+                "remarks:        **********************************************************\n" +
+                "remarks:        * This is a placeholder object to protect personal data.\n" +
+                "remarks:        * To view the original object, please query the RIPE\n" +
+                "remarks:        * Database at:\n" +
+                "remarks:        * http://www.ripe.net/whois\n" +
+                "remarks:        **********************************************************\n" +
+                "created:        2009-07-24T17:00:00Z\n" +
+                "last-modified:  2009-07-24T17:00:00Z\n" +
+                "source:         RIPE"
+        );
+    }
+    private void processMetadata(String source, UpdateNotificationFileResponse updateNotificationFile, String firstRecord) {
+        final JSONObject jsonObject = new JSONObject(firstRecord);
+        final int version = jsonObject.getInt("version");
+        final String sessionId = jsonObject.getString("session_id");
+        if (!sessionId.equals(updateNotificationFile.getSessionID())) {
+            // TODO: [MH] if the service is wrong for any reason...we have here a non-ending loop, we need to
```

Comment:
Resolve TODO

Reply:
I don't know how to resolve it. I put this here to discuss with you